hikari/impl/rest.py

```diff
@@ -133,6 +134,115 @@ def acquire(self) -> aiohttp.BaseConnector:
         return self.connector
 
 
+class ClientCredentialsStrategy(rest_api.TokenStrategy):
```

Comment:
I initially thought that this would handle authorizing and refreshing user tokens through OAuth. I think I'm confused cause this seem to do something that I have not seen before nor can find documented on the API docs. Would you mind explaining it to me?

Also, if this indeed doesn't handle user OAuth, would a class for that be worth implementing or we re-add the comment to use `aiohttp-oauth`?

Reply:
That's the client credentials code grant and its very well documented. This pr introduces all the logic needed for authenticating and refreshing oauth2 access grant, extended bot auth, client credentials, and access tokens and won't handle stuff like storing the token in a database and providing an auth page for initiating the oauth2 grant since those are out of scope and highly dependent on the env of the project being developed. The token strategy is barely an oauth2 feature, it's initial intended use is to allow a rest client to make requests as an application while authorising using the client credentials grant while regenerating said auth whenever negating the need to rebuild the whole client and was designed to be flexible enough to let users slot in their own logic for generating the token as they please; this is a feature that's primarily intended for bot-less slash commands.

Reply:
Oh, nice. First time I see that page in the documentation :sweat_smile: